docs/milestone-2.md

```diff
@@ -0,0 +1,297 @@
+3. mev-boost calls the [`relay_getValidatorStatusV1`](#relay_getvalidatorstatusv1) method on the relays to validate that the fee recipient is correctly registered.
+
+### Block Proposal
+
+```sequence
+participant consensus
+participant mev_boost
+participant execution
+participant relays
+Title: Block Proposal
+Note over consensus: wait for allocated slot
+consensus->mev_boost: engine_forkchoiceUpdatedV1
+mev_boost->execution: engine_forkchoiceUpdatedV1
+Note over mev_boost: begin polling
+mev_boost->relays: relay_getValidatorStatusV1
+mev_boost->relays: relay_getPayloadHeaderV1
+consensus->mev_boost: builder_getPayloadHeaderV1
+Note over mev_boost: select best payload
+mev_boost->execution: engine_getPayloadV1
+mev_boost-->consensus: builder_getPayloadHeaderV1 response
+Note over consensus: sign the block
+consensus->mev_boost: builder_proposeBlindedBlockV1
+Note over mev_boost: identify payload source
+mev_boost->relays: relay_proposeBlindedBlockV1
+Note over relays: validate signature
+relays-->mev_boost: relay_proposeBlindedBlockV1 response
+mev_boost-->consensus: builder_proposeBlindedBlockV1 response
+```
+
+1. mev-boost must be initialized with a list of (`BLSPublicKey`, `RelayURI`) pairs representing trusted relays.
+2. mev-boost must intercept [`engine_forkchoiceUpdatedV1`](#engine_forkchoiceupdatedv1) call from the BN -> EC and begin the following polling cycle:
+    1. mev-boost verifies that the relay has the correct [`FeeRecipient`](#feerecipient) for this slot tenant by calling [`relay_getValidatorStatusV1`](#relay_getvalidatorstatusv1) on the relay, if the incorrect fee recipient is returned, mev-boost enters the [`Authentication`](#authentication) process.
+    2. mev-boost checks for a fraud proof by calling [`relay_getRelayStatusV1`](#relay_getrelaystatusv1) on all connected relays. If a fraud proof is returned, mev-boost preforms the [`Validation`](#validation) process.
+    3. mev-boost begins polling connected relays for their [`SignedMEVPayloadHeader`](#signedmevpayloadheader) using [`relay_getPayloadHeaderV1`](#relay_getpayloadheaderv1) requests.
+    4. mev-boost must verify the signature and the merkle proof of the [`SignedMEVPayloadHeader`](#signedmevpayloadheader) received to ensure the signature matches the BLS key associated with the IP of the relay, has a matching `payloadId`, and has a valid fee recipient payment.
+5. upon receiving a [`builder_getPayloadHeaderV1`](#builder_getpayloadheaderv1) request from the BN, mev-boost must return the [`ExecutionPayloadHeaderV1`](https://github.com/ethereum/consensus-specs/blob/v1.1.6/specs/merge/beacon-chain.md#executionpayloadheader) with the highest associated `feeRecipientBalance` and terminate the polling cycle. If no eligible payload is received from a relay, mev-boost must request and return a payload from the local execution client using [`engine_getPayloadV1`](#engine_getpayloadv1).
```

Comment:
> If no eligible payload is received from a relay, mev-boost must request and return a payload from the local execution client using [`engine_getPayloadV1`](#engine_getpayloadv1).
	
I have a strong preference for fallback logic like this to be solely the responsibility of the consensus client. 

Fallback logic here would necessitate mev-boost getting a full execution payload from the execution client, translating it to a blinded payload, communicating it to the consensus client, which signs and returns a blinded block, translating it back to a full signed beacon block and then publishing it via the relay... when we could just use the normal proposal flow instead (and send the block directly to p2p). I think this would be much less likely to result in a missed proposal.

Consensus clients will likely implement fallback-from-mev-boost logic regardless, so I think this endpoint should just return a status code indicating a failure as quickly as possible, and let the consensus client can handle it.

Reply:
> I have a strong preference for fallback logic like this to be solely the responsibility of the consensus client.

This is the case in the current design, where mev-boost doesn't talk with the EL clients anymore: https://raw.githubusercontent.com/flashbots/mev-boost/main/docs/mev-boost-integration-overview.png